Px.Search.Lucene/LuceneSearcher.cs

```diff
@@ -46,25 +38,53 @@ public LuceneSearcher(string indexDirectory, string language)
         /// <param name="pageSize"></param>
         /// <param name="pageNumber"></param>
         /// <returns></returns>
-        public IEnumerable<SearchResult> Find(string searchExpression, int pageSize, int pageNumber)
+        public SearchResultContainer Find(string? query, int pageSize, int pageNumber, int? pastdays, bool includediscontinued = false)
         {
-            // See https://github.com/statisticssweden/Px.Search.Lucene/blob/main/Px.Search.Lucene/LuceneSearcher.cs
 
             var skipRecords = pageSize * (pageNumber - 1);
-            List<SearchResult> searchResultList = new List<SearchResult>();
+            var searchResultContainer = new SearchResultContainer();
+            var searchResultList = new List<SearchResult>();
             string[] fields = GetSearchFields();
             LuceneVersion luceneVersion = LuceneVersion.LUCENE_48;
-
-            QueryParser qp = new MultiFieldQueryParser(luceneVersion,
+            Query luceneQuery;
+            QueryParser queryParser = new MultiFieldQueryParser(luceneVersion,
                                                        fields,
                                                        new StandardAnalyzer(luceneVersion));
-            qp.DefaultOperator = _defaultOperator;
-            Query q = qp.Parse(searchExpression);
-            TopDocs topDocs = _indexSearcher.Search(q,skipRecords+pageSize);
+            BooleanFilter filter = new BooleanFilter();
+            queryParser.DefaultOperator = _defaultOperator;
+
+            luceneQuery = string.IsNullOrEmpty(query) ? queryParser.Parse("*:*"): queryParser.Parse(query);
+            
+            if (!string.IsNullOrEmpty(pastdays.ToString()))
+            {
+                var pastDay = DateTools.DateToString(DateTime.Now.AddDays(- Convert.ToDouble(pastdays)), DateResolution.HOUR);
+                filter.Add(new FilterClause(FieldCacheRangeFilter.NewStringRange(SearchConstants.SEARCH_FIELD_UPDATED, lowerVal: pastDay, includeLower: true, upperVal: null, includeUpper: false), Occur.MUST));
+            }
+            
+            if (!includediscontinued)
+            {
+                var areaFilter = new TermsFilter(new Term(SearchConstants.SEARCH_FIELD_DISCONTINUED, "false"));
+                filter.Add(new FilterClause(areaFilter, Occur.MUST));
+            }
+
+            TopDocs topDocs;
+            topDocs = filter.Count() > 0 ? _indexSearcher.Search(luceneQuery, filter, skipRecords + pageSize) 
+                : _indexSearcher.Search(luceneQuery, skipRecords + pageSize);
+
             ScoreDoc[] scoreDocs = topDocs.ScoreDocs;
             DateTime updated;
             bool discontinued;
+            var pages = (double)topDocs.TotalHits / (double)pageSize;
+            int totalpages;
 
+            totalpages = IsInteger(pages) ? (int)pages : (int)pages + 1;
```

Comment:
I think that you might do  `totalPages = Convert.ToInt32(Math.Ceiling(pages));` instead